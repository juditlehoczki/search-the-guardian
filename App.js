import React, { Component } from "react";
import {
  ScrollView,
  View,
  Text,
  Linking,
  FlatList,
  TouchableOpacity,
} from "react-native";

import { fetchArticles } from "./api.js";
import Card from "./components/ArticleCard.js";
import Categories from "./components/Categories.js";
import Paginator from "./components/Paginator.js";
import Header from "./components/Header.js";
import Search from "./components/Search.js";
import Subscribe from "./components/Subscribe.js";

import s from "./styles.js";

import { YellowBox } from "react-native";
YellowBox.ignoreWarnings([
  "VirtualizedLists should never be nested", // TODO: Remove when fixed
]);

export default class App extends Component {
  state = {
    searchInput: "",
    lastSearch: "",
    lastSection: "",
    fiveNewestArticles: [],
    articles: [],
    resultsCount: 0,
    currentPage: 1,
    pagesCount: 0,
    articlesNotYetRequested: true,
  };

  componentDidMount() {
    this.getFiveNewestArticles();
  }

  getFiveNewestArticles() {
    fetchArticles(1, null, null, "newest", 5).then((res) => {
      const { results } = res.data.response;
      this.setState({
        fiveNewestArticles: results,
      });
    });
  }

  componentDidUpdate(prevProps, prevState) {
    this.scroll.scrollTo({
      x: 0,
      y: 0,
      animated: true,
    });
    const { currentPage, lastSearch, lastSection } = this.state;
    if (prevState.currentPage !== currentPage) {
      if (lastSearch) {
        this.getArticles(currentPage, lastSearch);
      }
      if (lastSection) {
        this.getArticlesBySection(currentPage, lastSection);
      }
    }
  }

  handleChange(value) {
    this.setState({
      searchInput: value,
    });
  }

  changePage(change) {
    this.setState({
      currentPage: this.state.currentPage + change,
    });
  }

  refreshMainPage() {
    this.setState({
      articlesNotYetRequested: true,
    });
  }

  getArticles(page, keyword) {
    fetchArticles(page, keyword).then((res) => {
      const { results, total, currentPage, pages } = res.data.response;
      this.setState({
        articles: results,
        resultsCount: total,
        currentPage: currentPage,
        pagesCount: pages,
        articlesNotYetRequested: false,
        lastSection: "",
        lastSearch: keyword,
        searchInput: "",
      });
    });
  }

  getArticlesBySection(page, section) {
    fetchArticles(page, null, section, "newest", 10).then((res) => {
      const { results, total, currentPage, pages } = res.data.response;
      this.setState({
        articles: results,
        resultsCount: total,
        currentPage: currentPage,
        pagesCount: pages,
        articlesNotYetRequested: false,
        lastSearch: "",
        lastSection: section,
      });
    });
  }

  render() {
    const {
      searchInput,
      articlesNotYetRequested,
      articles,
      fiveNewestArticles,
      currentPage,
      pagesCount,
      resultsCount,
    } = this.state;

    return (
      <ScrollView
        ref={(top) => {
          this.scroll = top;
        }}
        contentContainerStyle={s.body}
        keyboardShouldPersistTaps="handled"
        contentInsetAdjustmentBehavior="automatic"
      >
        <View>
          <Header refreshMainPage={this.refreshMainPage.bind(this)} />
          <Search
            handleChange={this.handleChange.bind(this)}
            searchInput={searchInput}
            getArticles={this.getArticles.bind(this)}
          />
          {articlesNotYetRequested ? (
            <View>
              <Text style={s.sectionTitle}>Most Recent:</Text>
              <FlatList
                data={fiveNewestArticles}
                renderItem={({ item }) => {
                  return <Card result={item} />;
                }}
              />
              <Categories
                getArticlesBySection={this.getArticlesBySection.bind(this)}
              />
            </View>
          ) : (
            <View>
              <Text style={s.sectionTitle}>
                A total of {resultsCount} articles found.
              </Text>
              <Paginator
                currentPage={currentPage}
                pagesCount={pagesCount}
                changePage={this.changePage.bind(this)}
              />
              <View style={{ width: "100%" }}>
                <FlatList
                  data={articles}
                  renderItem={({ item }) => {
                    return <Card result={item} />;
                  }}
                />
              </View>
              <Paginator
                currentPage={currentPage}
                pagesCount={pagesCount}
                changePage={this.changePage.bind(this)}
              />
            </View>
          )}
        </View>
        <Subscribe />
        <TouchableOpacity>
          <Text
            style={s.credit}
            onPress={() => {
              Linking.openURL("http://juditlehoczki.me");
            }}
          >
            Developed With ðŸ’› By Judit Lehoczki
          </Text>
        </TouchableOpacity>
      </ScrollView>
    );
  }
}
