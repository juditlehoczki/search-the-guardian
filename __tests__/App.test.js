import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import axios from 'axios';
import {mockResponse} from './testData/apiResponseMock.js';

import App from '../App.js';

jest.mock('axios');
axios.get.mockImplementation(() => Promise.resolve(mockResponse));

describe('App Component', () => {
  test('Snapshot - it renders correctly', () => {
    const appSnapShot = renderer.create(<App />).toJSON();
    expect(appSnapShot).toMatchSnapshot();
  });
  test('componentDidMount puts 5 articles in the fiveNewestArticles state', async () => {
    let mockApp = renderer.create(<App />).getInstance();
    await mockApp.componentDidMount;
    expect(mockApp.state.fiveNewestArticles.length).toEqual(5);
  });
  test("handleChange method sets the searchInput state based on the argument it's called with", () => {
    let mockApp = renderer.create(<App />).getInstance();
    mockApp.handleChange('cycling');
    expect(mockApp.state.searchInput).toEqual('cycling');
  });
  test('changePage method works as expected - sets the currentPage state correctly', () => {
    let mockApp = renderer.create(<App />).getInstance();
    mockApp.changePage(1);
    expect(mockApp.state.currentPage).toEqual(2);
    mockApp.changePage(-1);
    expect(mockApp.state.currentPage).toEqual(1);
  });
  test('getArticles puts articles in the articles state', async () => {
    let mockApp = renderer.create(<App />).getInstance();
    await mockApp.getArticles(1, 'Cycling');
    expect(mockApp.state.articles.length).toEqual(5);
  });
  test('getArticlesBySection puts articles in the articles state', async () => {
    let mockApp = renderer.create(<App />).getInstance();
    await mockApp.getArticlesBySection('news');
    expect(mockApp.state.articles.length).toEqual(5);
  });
  test('refreshMainPage sets the articlesNotYetRequested state to true', () => {
    let mockApp = renderer.create(<App />).getInstance();
    mockApp.state.articlesNotYetRequested = false;
    mockApp.refreshMainPage();
    expect(mockApp.state.articlesNotYetRequested).toEqual(true);
  });
});
