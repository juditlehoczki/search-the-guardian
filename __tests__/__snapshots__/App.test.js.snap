// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot - it renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "bottom": 30,
    }
  }
  contentInsetAdjustmentBehavior="automatic"
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "#54C1B7",
              "fontSize": 20,
              "fontWeight": "bold",
              "letterSpacing": 10,
              "marginBottom": -27,
              "marginTop": 60,
              "textAlign": "center",
            }
          }
        >
          search
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "paddingVertical": 10,
                "resizeMode": "contain",
                "width": "95%",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginLeft": 5,
            }
          }
        >
          üîç
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Search..."
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#D1529E",
              "borderRadius": 5,
              "borderStyle": "dotted",
              "borderWidth": 1,
              "color": "#0F406E",
              "flex": 8,
              "marginLeft": 5,
              "marginVertical": 5,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D1529E",
              "borderRadius": 15,
              "flex": 1,
              "justifyContent": "center",
              "margin": 5,
              "opacity": 1,
              "padding": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
            Go!
          </Text>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "backgroundColor": "#54C1B7",
              "color": "white",
              "fontWeight": "bold",
              "marginVertical": 15,
              "paddingLeft": 15,
              "paddingVertical": 5,
            }
          }
        >
          Most Recent:
        </Text>
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "#54C1B7",
                "color": "white",
                "fontWeight": "bold",
                "marginVertical": 15,
                "paddingLeft": 15,
                "paddingVertical": 5,
              }
            }
          >
            Not sure what to read? Here are our favourite topics:
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                CULTURE
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                ENVIRONMENT
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                FILM
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                FOOD
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                MUSIC
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                SCIENCE
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                TECHNOLOGY
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 5,
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                  "marginVertical": 3,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#0F406E",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 3,
                    "padding": 10,
                  }
                }
              >
                TRAVEL
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#D1529E",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#54C1B7",
                    "fontSize": 18,
                  }
                }
              >
                &gt;
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "#54C1B7",
            "color": "white",
            "fontWeight": "bold",
            "marginVertical": 15,
            "paddingLeft": 15,
            "paddingVertical": 5,
          }
        }
      >
        Want more? Subscribe to our weekly newsletter:
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginLeft": 5,
            }
          }
        >
          ‚úâÔ∏è
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Email address..."
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#D1529E",
              "borderRadius": 5,
              "borderStyle": "dotted",
              "borderWidth": 1,
              "color": "#0F406E",
              "flex": 8,
              "marginLeft": 5,
              "marginVertical": 5,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D1529E",
              "borderRadius": 15,
              "flex": 1,
              "justifyContent": "center",
              "margin": 5,
              "opacity": 1,
              "padding": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
            Go!
          </Text>
        </View>
      </View>
    </View>
    <Text
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#D1529E",
          "color": "white",
          "fontWeight": "bold",
          "marginVertical": 15,
          "paddingRight": 15,
          "paddingVertical": 5,
          "textAlign": "right",
        }
      }
    >
      Developed With üíõ By Judit Lehoczki
    </Text>
  </View>
</RCTScrollView>
`;
